rust-connectors/sources/mqtt/src/main.rs

```diff
@@ -84,11 +94,32 @@ fn main() -> Result<(), MqttConnectorError> {
             %mqtt_topic,
             %client_id
         );
+        let port = if opts.mqtt_tls { 8883 } else { 1883 };
```

Comment:
I think that we should add a `mqtt_port` configuration just in case it is using a custom port.